solidity/test/token/EverclearTokenBridge.t.sol

```diff
@@ -0,0 +1,552 @@
+}
+
+contract EverclearTokenBridgeTest is Test {
+    using TypeCasts for address;
+
+    // Constants
+    uint32 internal constant ORIGIN = 11;
+    uint32 internal constant DESTINATION = 12;
+    uint8 internal constant DECIMALS = 18;
+    uint256 internal constant SCALE = 1e18;
+    uint256 internal constant TOTAL_SUPPLY = 1_000_000e18;
+    uint256 internal constant TRANSFER_AMT = 100e18;
+    uint256 internal constant FEE_AMOUNT = 5e18; // 5 tokens fee
+    uint256 internal constant GAS_PAYMENT = 0.001 ether;
+    string internal constant NAME = "TestToken";
+    string internal constant SYMBOL = "TT";
+
+    // Test addresses
+    address internal constant ALICE = address(0x1);
+    address internal constant BOB = address(0x2);
+    address internal constant OWNER = address(0x3);
+    address internal constant PROXY_ADMIN = address(0x37);
+
+    // Mock contracts
+    ERC20Test internal token;
+    MockMailbox internal mailbox;
+    MockEverclearAdapter internal everclearAdapter;
+    TestPostDispatchHook internal hook;
+
+    // Main contract
+    EverclearTokenBridge internal bridge;
+
+    // Test data
+    bytes32 internal constant OUTPUT_ASSET = bytes32(uint256(0x456));
+    bytes32 internal constant RECIPIENT = bytes32(uint256(uint160(BOB)));
+    uint256 internal feeDeadline;
+    bytes internal feeSignature = hex"1234567890abcdef";
+
+    // Events to test
+    event FeeParamsUpdated(uint256 fee, uint256 deadline);
+    event OutputAssetSet(uint32 destination, bytes32 outputAsset);
+
+    function setUp() public {
+        // Setup basic infrastructure
+        mailbox = new MockMailbox(ORIGIN);
+        token = new ERC20Test(NAME, SYMBOL, TOTAL_SUPPLY, DECIMALS);
+        everclearAdapter = new MockEverclearAdapter();
+        hook = new TestPostDispatchHook();
+
+        // Set fee deadline to future
+        feeDeadline = block.timestamp + 3600; // 1 hour from now
+
+        // Deploy bridge implementation
+        EverclearTokenBridge implementation = new EverclearTokenBridge(
+            address(token),
+            SCALE,
+            address(mailbox),
+            everclearAdapter
+        );
+
+        // Deploy proxy
+        TransparentUpgradeableProxy proxy = new TransparentUpgradeableProxy(
+            address(implementation),
+            PROXY_ADMIN,
+            abi.encodeWithSelector(
+                EverclearTokenBridge.initialize.selector,
+                address(hook),
+                address(0), // No ISM
+                OWNER
+            )
+        );
+
+        bridge = EverclearTokenBridge(address(proxy));
+
+        // Setup initial state
+        vm.startPrank(OWNER);
+        bridge.setFeeParams(FEE_AMOUNT, feeDeadline, feeSignature);
+        bridge.setOutputAsset(DESTINATION, OUTPUT_ASSET);
+        // Enroll remote router for destination domain => Only needed for `quoteTransferRemote`
+        bridge.enrollRemoteRouter(DESTINATION, OUTPUT_ASSET); // Using OUTPUT_ASSET as mock router address
```

Comment:
if the router behavior is not used at all maybe we should remove

Reply:
Removed